### PR TITLE
Add doc and prodsec approval basic function

### PR DESCRIPTION
**Pull request by wewang58**
@rioliu-rh please check it, thanks

### CODE REVIEW

File: oar/core/notification_mgr.py [hunk 1]
```diff
@@ -151,8 +151,24 @@ def share_dropped_and_must_verify_bugs(self, dropped_bugs, must_verify_bugs):
             raise NotificationException(
                 "share dropped and must verified bugs failed"
             ) from e
-
-
+    
+    def share_request_doc_secu_approval_result(self, doc_appr, prodsec_appr):
```
Comment (rioliu-rh) [inline oar/core/notification_mgr.py:155]:
```suggestion
    def share_request_doc_prodsec_approval_result(self, doc_appr, prodsec_appr):
```

File: oar/core/notification_mgr.py [hunk 2]
```diff
@@ -443,6 +459,45 @@ def get_slack_message_for_dropped_and_must_verify_bugs(
 
         return message
 
+    def get_slack_message_for_docs_and_prodsec_approve(
```
Comment (rioliu-rh) [inline oar/core/notification_mgr.py:462]:
```suggestion
    def get_slack_message_for_docs_and_prodsec_approval(
```

File: oar/core/notification_mgr.py [hunk 3]
```diff
@@ -443,6 +459,45 @@ def get_slack_message_for_dropped_and_must_verify_bugs(
...
+        gid = self.sc.get_group_id_by_name(
+            self.cs.get_slack_user_group_from_contact(
+                "art", "doc_id"
+            )
+        )
+        userid = self.cs.get_slack_user_id_from_email("art","secu_email")
+        message = ""
+        if len(doc_appr)&len(prodsec_appr):
+            logger.info(
+                f"notfiy for Docs and Prodsec approval"
```
Comment (rioliu-rh) [inline oar/core/notification_mgr.py:484]:
```suggestion
                f"send message for Docs and Prodsec approval"
```

File: oar/core/config_store.json [hunk 4]
```diff
@@ -16,8 +16,10 @@
     "contacts": {
         "slack": {
             "art": {
-                "channel": "#forum-release",
-                "id": "release-artists"
+                "channel": "#forum-ocp-release",
+                "id": "release-artists",
+                "doc_id": "docs-rel-notes",
+                "secu_email" : "ahanwate@redhat.com"
```
[the remark this answers is not in the document]
Comment (rioliu-rh) [reply oar/core/config_store.json:22]: and I think prodsec contact is not only one.

File: oar/core/notification_mgr.py [hunk 5]
```diff
@@ -443,6 +459,45 @@ def get_slack_message_for_dropped_and_must_verify_bugs(
...
+                )
+            message = f"[{self.cs.release}] Hello {gid}, please approve Docs for advisories:{str(doc_appr)} and {userid} Prodsec for advisories:{str(prodsec_appr)}, thanks!"
```
Comment (rioliu-rh) [inline oar/core/notification_mgr.py:486]: it's better to add z-stream release version at the beginning. let's avoid using `Please` , instead `Could you approve xxx`

File: oar/core/advisory_mgr.py [hunk 6]
```diff
@@ -172,6 +172,22 @@ def change_advisory_status(self, status=AD_STATUS_REL_PREP):
         except Exception as e:
             raise AdvisoryException(f"change advisory status failed") from e
 
+    def get_requested_doc_security_ads(self,ads):
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:175]: can you add a unit test for this func

File: oar/cli/cmd_change_advisory_status.py [hunk 7]
```diff
@@ -3,7 +3,6 @@
 from oar.core.worksheet_mgr import WorksheetManager
 from oar.core.advisory_mgr import AdvisoryManager
 from oar.core.jira_mgr import JiraManager
-from oar.core.config_store import ConfigStore
```
Comment (rioliu-rh) [inline oar/cli/cmd_change_advisory_status.py:6]: what's the purpose about this change?

File: oar/core/advisory_mgr.py [hunk 8]
```diff
@@ -172,6 +172,22 @@ def change_advisory_status(self, status=AD_STATUS_REL_PREP):
...
+        """
+        get requested  Docs and product security approval advisories
+        """
+        try:
+            doc_appr = []
+            security_appr = []
+            for ad in ads:
+                if ad.get_erratum_data()["doc_complete"] == 0:
+                    doc_appr.append(ad.errata_id)
+                if ad.get_erratum_data()["security_approved"] == False:
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:185]: is it possible that the value of `security_approved ` is null
```
advisory[security_approved]: null for security approval not requested, false for requested, true for approved
```

File: oar/core/notification_mgr.py [hunk 9]
```diff
@@ -443,6 +459,45 @@ def get_slack_message_for_dropped_and_must_verify_bugs(
...
+                f"notfiy for doc approval"
+                ) 
+            message = f"[{self.cs.release}] Hello {gid}, please approve doc for advisories:{str(doc_appr)}, thanks!"
+        elif len(prodsec_appr):
+            logger.info(
+                f"notfiy for Prodsec approval"
+                ) 
+            message = f"[{self.cs.release}] Hello {userid}, please approve Prodsec for advisories:{str(prodsec_appr)}, thanks!"
+        else:
+            logger.info("no advisory need to doc or Prodsec approval")
```
Comment (rioliu-rh) [inline oar/core/notification_mgr.py:498]:
```suggestion
            logger.info("no advisory need Doc or Prodsec approval")
```

File: oar/core/advisory_mgr.py (hunk 6 above)
Comment (wewang58) [reply oar/core/advisory_mgr.py:175]: Added, test result: 
[wewang@ovpn-12-217 tests]$   python3 -m unittest test_advisory_mgr.TestAdvisoryManager.test_get_requested_doc_security_ads
.
----------------------------------------------------------------------
Ran 1 test in 23.390s

OK

Comment (wewang58) [reply oar/core/advisory_mgr.py:175]: add test_get_requested_doc_security_ads:
[wewang@ovpn-12-217 tests]$   python3 -m unittest test_advisory_mgr.TestAdvisoryManager.test_get_requested_doc_security_ads
.
----------------------------------------------------------------------
Ran 1 test in 23.390s

OK

File: oar/cli/cmd_change_advisory_status.py (hunk 7 above)
Comment (wewang58) [reply oar/cli/cmd_change_advisory_status.py:6]: it's not used, so delete it

File: oar/core/advisory_mgr.py (hunk 8 above)
Comment (wewang58) [reply oar/core/advisory_mgr.py:185]: if it's null or true, will nothing to do.

Comment (rioliu-rh) [reply oar/core/advisory_mgr.py:185]: a little confused here, `doc_complete` is not used to check doc approval is requested or not. 
```
advisory[doc_complete]: '1' to approve docs, '0' to disapprove docs
```
another question, why we get approval requested advisories here, we should get approved advisories.

**Comment by wewang58** [comment]
@rioliu-rh  Updated according to our discussion, about notify part in command drop bug part, when has requirement and has errata to test, will add it later, thanks

File: oar/core/advisory_mgr.py [hunk 10]
```diff
@@ -277,6 +277,22 @@ def check_cve_tracker_bug(self):
 
         return cve_tracker_bugs
 
+    def get_doc_security_approved_ads(self):
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:280]:
```suggestion
    def get_doc_prodsec_approved_ads(self):
```

File: oar/core/advisory_mgr.py [hunk 11]
```diff
@@ -397,3 +413,46 @@ def are_all_push_jobs_completed(self):
...
+    def is_doc_approved(self):
+        """
+        Check if doc is approved for a advisory
+        Returns:
+        bool: True if doc for a advisory is approved, otherwise False
+        """
+        appr_status = self.get_erratum_data()["doc_complete"]
+        if appr_status == 1:
+            return True
+        return False
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:426]:
```suggestion
        return appr_status == 1
```

File: oar/core/advisory_mgr.py [hunk 12]
```diff
@@ -397,3 +413,46 @@ def are_all_push_jobs_completed(self):
...
+    def is_doc_requested(self):
+        """
+        Check if doc for a advisory is requested 
+        Returns:
+        bool: True if doc for a advisory is requested, otherwise False        
+        """
+        request_status = self.get_erratum_data()["text_ready"]
+        if request_status == 1:
+            return True
+        return False
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:446]:
```suggestion
        return request_status == 1
```

File: oar/core/advisory_mgr.py [hunk 13]
```diff
@@ -397,3 +413,46 @@ def are_all_push_jobs_completed(self):
...
+        Check if doc is approved for a advisory
+        Returns:
+        bool: True if doc for a advisory is approved, otherwise False
+        """
+        appr_status = self.get_erratum_data()["doc_complete"]
+        if appr_status == 1:
+            return True
+        return False
+    
+    def is_prodsec_approved_or_requested(self):
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:428]: suggest to split `approved` and `requested` to different funcs

File: oar/core/advisory_mgr.py [hunk 14]
```diff
@@ -397,3 +413,46 @@ def are_all_push_jobs_completed(self):
...
+
+    def request_doc_approval(self):
+        """
+        send request for doc approval for a advisory
+        """
+        self.addFlags('needs_docs') 
+           
+    def request_prodsec_approval(self):
+        """
+        send product security request for doc approval for a advisory
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:456]:
```suggestion
        send product security approval request for a advisory
```

File: oar/core/advisory_mgr.py [hunk 15]
```diff
@@ -397,3 +413,46 @@ def are_all_push_jobs_completed(self):
...
+        bool: True if doc for a advisory is requested, otherwise False        
+        """
+        request_status = self.get_erratum_data()["text_ready"]
+        if request_status == 1:
+            return True
+        return False
+
+    def request_doc_approval(self):
+        """
+        send request for doc approval for a advisory
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:450]:
```suggestion
        send doc approval request for a advisory
```

File: oar/core/advisory_mgr.py (hunk 13 above)
Comment (wewang58) [reply oar/core/advisory_mgr.py]: could you check again, thanks

File: oar/core/advisory_mgr.py [hunk 16]
```diff
@@ -397,3 +413,51 @@ def are_all_push_jobs_completed(self):
...
         return completed
+
+    def is_doc_approved(self):
+        """
+        Check if doc is approved for a advisory
+        Returns:
+        bool: True if doc for a advisory is approved, otherwise False
+        """
+        appr_status = self.get_erratum_data()["doc_complete"]
+        return appr_status == 1
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:424]:
```suggestion
        return self.get_erratum_data()["doc_complete"] == 1
```

File: oar/core/advisory_mgr.py [hunk 17]
```diff
@@ -277,6 +277,22 @@ def check_cve_tracker_bug(self):
...
+        get Docs and product security approved advisories
+        """
+        try:
+            approved_doc_ads = []
+            approved_prodsec_ads = []
+            ads = self.get_advisories()
+            for ad in ads:
+                if ad.is_doc_approved():
+                    approved_doc_ads.append(ad)
+                if ad.is_prodsec_approved():
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:291]:
```suggestion
                if ad.errata_type == "RHSA" and ad.is_prodsec_approved():
```

File: oar/core/advisory_mgr.py [hunk 18]
```diff
@@ -397,3 +413,51 @@ def are_all_push_jobs_completed(self):
...
+        appr_status = self.get_erratum_data()["doc_complete"]
+        return appr_status == 1
+    
+    def is_prodsec_approved(self):
+        """
+        Check if prodsec is approved for a advisory
+        Returns:
+        bool: True if prodsec is approved, otherwise False
+        """
+        return self.get_erratum_data()["security_approved"]
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:432]:
```suggestion
        return self.get_erratum_data()["security_approved"] == True
```

File: oar/core/advisory_mgr.py [hunk 19]
```diff
@@ -397,3 +413,51 @@ def are_all_push_jobs_completed(self):
...
+           
+
+    def is_doc_requested(self):
+        """
+        Check if doc for a advisory is requested 
+        Returns:
+        bool: True if doc for a advisory is requested, otherwise False        
+        """
+        request_status = self.get_erratum_data()["text_ready"]
+        return request_status == 1
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:442]:
```suggestion
        return self.get_erratum_data()["text_ready"] == 1
```

File: oar/core/advisory_mgr.py [hunk 20]
```diff
@@ -397,3 +413,51 @@ def are_all_push_jobs_completed(self):
...
+        bool: False if prodsec for a advisory is requested, otherwise False        
+        """
+        return self.get_erratum_data()["security_approved"] == False
+
+
+    def request_doc_approval(self):
+        """
+        send doc approval request for a advisory
+        """
+        self.addFlags('needs_docs')
```
[the remark this answers is not in the document]
Comment (rioliu-rh) [reply oar/core/advisory_mgr.py:456]: flag won't update advisory. need to update `doc_complete`

File: oar/core/advisory_mgr.py [hunk 21]
```diff
@@ -397,3 +413,49 @@ def are_all_push_jobs_completed(self):
...
+        """
+        send doc approval request for a advisory
+        """
+        self.addFlags('needs_docs') 
+           
+    def request_prodsec_approval(self):
+        """
+        send product security approval request for a advisory
+        """
+        self.addFlags('request_security')
```
Comment (rioliu-rh) [inline oar/core/advisory_mgr.py:461]: same concern like above, need to update `security_approved`

File: oar/core/advisory_mgr.py (hunk 20 above)
[the remark this answers is not in the document]
Comment (wewang58) [reply oar/core/advisory_mgr.py:455]: try to find  a available advisory to check

**Pull request merged**